src/containers/DataPage/DataPage.jsx

```diff
@@ -390,21 +394,99 @@ class DataPage extends PureComponent {
     );
   }
 
-  onStartDownload() {
-    const { dispatch } = this.props;
-    dispatch(DataPageActions.changeDownloadStatus('started'));
-  }
+  getDownloadCombinations() {
+    function makeEntry(id, type, infos) {
+      const info = infos.find(d => d.id === id) || { label: id };
+      return {
+        [`${type}Id`]: id,
+        [`${type}Label`]: info.label,
+      };
+    }
 
-  renderDownloadResults() {
-    const { downloadStatus } = this.props;
+    function labelFromTypes(types) {
+      const labelMap = {
+        clientIsp: 'Client ISPs',
+        transitIsp: 'Transit ISPs',
+        location: 'Locations',
+      };
+      const separator = ' --- ';
 
-    if (!downloadStatus) {
-      return null;
+      return types.map(type => labelMap[type]).join(separator);
     }
 
+    return facetTypeCombinations.filter(types => {
+      // only keep the combination types for which we have items for
+      const hasEntries = types.every(type => this.props[`${type}Ids`].length);
+      return hasEntries;
+    // build up all combinations for the types we have items for
+    }).map(combinationTypes => {
+      let entries = [];
+      combinationTypes.forEach(partType => {
```

Comment:
this inner foreach is a bit complex - and could be moved to a separate function for readability perhaps.